README.md

```diff
@@ -188,7 +158,119 @@ make run
 
 You should be able to reach the service at: http://0.0.0.0:8080. The port that is used is controlled by an environment variable defined in the `.flaskenv` file which Flask uses to load it's configuration from the environment by default.
 
-Test for webhook
+
+## Behavior-Driven Development (BDD)
+
+Behavior-Driven Development (BDD) is implemented to test the application from the end-user's perspective. It ensures that the service behaves as expected when accessed through its UI. Tests are written in Gherkin syntax and executed using `behave` with Selenium.
+
+---
+
+### BDD Features
+
+The BDD tests validate the following operations:
+1. **Create a Promotion**: Verify that new promotions can be added through the UI.
+2. **Read a Promotion**: Ensure that promotion details can be viewed.
+3. **Update a Promotion**: Test that existing promotions can be updated.
+4. **Delete a Promotion**: Confirm that promotions can be removed.
+5. **List Promotions**: Verify that all promotions are displayed in the UI.
+6. **Query Promotions**: Test the ability to filter promotions by specific criteria.
+7. **Action on Promotions**: Validate custom actions defined for promotions.
+
+---
+
+### Running BDD Tests
+
+To execute the BDD tests:
+
+   ```bash
+   behave
+   ```
+
+## Kubernetes Deployment
+
+This microservice is deployed to a Kubernetes cluster using the manifests provided in the `k8s/` directory.
+
+### Local Development Setup
```

Comment:
Do you mean Local Deployment Setup here?

Reply:
Yes I think you are correct. I have modify the wording in the new commit. Thanks!